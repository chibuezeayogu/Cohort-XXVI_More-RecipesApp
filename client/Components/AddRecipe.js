import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Button, Col, Table} from 'react-materialize';
import axios from 'axios';
import Preloader from './Preloder';
//require('dotenv').config();


const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;


class AddRecipe extends Component{

    constructor(props) {
        super(props);
        this.state = {
            file: '',
            imagePreviewUrl: '', 
            ImageURL: '',
            arrIngredients:[],
            arrProcedures:[]
        };
      }

      componentWillMount(){
        // if(!this.props.userData.isAuthenticated){
        //     this.context.router.push('/sign');
        // }
        const token = localStorage.getItem('jwtToken');
        if(!token){
            this.context.router.push('/signin');
        }
    }
    //   componentWillUpdate(nextProps, nextState) {
    //     console.log(nextState.arrIngredients);
    // }
   
      _addIngredients(e){
          e.preventDefault();
          const ingredients = document.getElementById('ingredients').value;
          if(!ingredients){
                alert("Ingredients field is empty");
          }else {

            const newIngredients = this.state.arrIngredients.concat(ingredients);
                this.setState({
                    arrIngredients: newIngredients
                });
            }
      }

     



      _addProcedures(e){
        e.preventDefault();
        const procedures = document.getElementById('procedures').value;
        if(!procedures){
              alert("Ingredients field is empty");
        }else {

          const newProcedures = this.state.arrProcedures.concat(procedures);
              this.setState({
                    arrProcedures: newProcedures
              })
          }
        
          this.ingredient.state.reset();
      }

      _handleImageChange(e) {
        e.preventDefault();
    
        let reader = new FileReader();
        let file = e.target.files[0];
        const formData = new FormData;
        formData.append('file', file);
        formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
        
        // if (!file.type.match('image.*')){

        // }else{

        // }
        
        $.ajax({
            type: "POST",
            url: CLOUDINARY_URL,
            dataType: 'json',
            data: formData,
            contentType:false,
            processData: false,
            success: ((data) => {
                this.setState({ImageURL:data.secure_url});
            })
        });

        reader.onloadend = () => {
          this.setState({
            file: file,
            imagePreviewUrl: reader.result
          });
        };
    
        reader.readAsDataURL(file);
      }
      
    handleOnsubmit(e) {
        e.preventDefault();
        const title = this.title.state.value;
        const description = this.description.state.value;
        const ingredients = this.ingredients.state.value;
        const procedures = this.procedures.state.value;
        const imageURL = this.state.ImageURL;
        this.props.addRecipe(title, description, ingredients, procedures, imageURL);
        this.refs.AddRecipeForm.reset();
        this.setState({
            imagePreviewUrl:''
        });
    }
    render() {
        // let {imagePreviewUrl} = this.state;
        // let $imagePreview = null;
        // if (imagePreviewUrl) {
        //   $imagePreview =(<img  class="responsive-img circle" src={imagePreviewUrl}/>);
        // } else {
        //   $imagePreview = " ";
        // }
       const  { arrIngredients, arrProcedures } = this.state;
        return (
            <div className="container"> 
                <h1 className="center">Add Recipe</h1>
                <hr/>
                <form encType="multipart/form-data" methos="post" className="col l6 m8 s12 offset-l8 offset-m6" onSubmit={(e) => this.handleOnsubmit(e)} ref="AddRecipeForm">
                    <Row>
                        <Input s={12} label="Title"  type='text' ref={(input)=> {this.title=input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row s={8}>
                        <Input s={12} label="Description" className='validate' type='text' ref={(input) => {this.description = input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={12} label="Ingredients" id="ingredients" type='text' ref={(input) => {this.ingredients = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addIngredients(e)}}>Add Ingredient</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrIngredients
                                    .map((ingredient, i) => 
                                        <tr>
                                            <td className="s10">
                                                <input id="ingredients" type='text' value={ingredient}></input>
                                            </td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={12} label="Procedures" id="procedures" type='text' ref={(input) => {this.procedures = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addProcedures(e)}}>Add Procedures</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrProcedures
                                    .map((procedures, i) => 
                                        <tr>
                                            <td className="s10">{procedures}</td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={6} id="recipe"  name="file" type='file' ref={(input) => {this.file = input}}   onChange={(e)=>this._handleImageChange(e)}><Icon>insert_photo</Icon></Input>
                        <Col s={6} ><img className="responsive-img img-wh z-depth-3" src={this.state.imagePreviewUrl && this.state.ImageURL?this.state.imagePreviewUrl:"select image to preview"}/></Col>
                    </Row>
                    <br />
                    <Row>
                        <Button waves='light' className='right'><Icon right>send</Icon>Add</Button>
                    </Row>
                </form>
            </div>
        );
    }
}

AddRecipe.contextTypes = {
    router: React.PropTypes.object
};


AddRecipe.propTypes = {
handleOnSubmit: PropTypes.func.isRequired,
_addIngredients:  PropTypes.func.isRequired,
_addProcedures:  PropTypes.func.isRequired,
_handleImageChange:  PropTypes.func.isRequired,
addRecipe: PropTypes.func.isRequired,
};

export default AddRecipe;