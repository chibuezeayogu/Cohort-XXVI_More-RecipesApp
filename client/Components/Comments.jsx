import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
