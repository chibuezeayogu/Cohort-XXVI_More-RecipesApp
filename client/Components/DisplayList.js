import React, { Component } from 'react';
import { Link } from 'react-router';

const Displaylist = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <td>
                list ingradients
            </td>
        );
    }
});

export default Displaylist;