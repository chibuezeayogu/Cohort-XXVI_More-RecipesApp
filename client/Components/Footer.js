import React, { Component }from 'react';
import PropTypes from 'prop-types';
import { Footer } from 'react-materialize';


class BottomFooter extends  Component {
    render() {
        return (
            <div>
                <Footer copyrights="&copy 2017 Copyright Text"
                moreLinks={
                    <a className="grey-text text-lighten-4 right" href="#!">Andela Project by Ayogu Chibueze></a>
                }
                
                className='black'
            >
                    
                </Footer>;
            </div>
        );
    }
}

BottomFooter.propTypes = {
    logout: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
    className: PropTypes.string,
    copyrights: PropTypes.string,
};

export default BottomFooter;