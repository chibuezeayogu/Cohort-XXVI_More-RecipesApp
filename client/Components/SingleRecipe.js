import React, { Component } from 'react';
import { Link } from 'react-router';

const SingleRecipe = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <div class="col s12 m2">
                <figure>
                i am the single
                </figure>
            </div>
        );
    }
});

export default SingleRecipe;