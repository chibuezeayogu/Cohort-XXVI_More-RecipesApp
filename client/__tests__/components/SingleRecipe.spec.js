
import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { SingleRecipe } from '../../Components/SingleRecipe';
import mockData from '../__mock__/actionMockData';

const props = {
  recipeReducer: {
    isFetched: true,
    recipes: mockData.recipe,
  },
  userData: { currentUser: { id: 1, email: 'chibuezeayogu@hotmail.com' } },
  commentReducer: {
    reviews: mockData.reviews
  },
  match: {
    params: {
      id: 1
    }
  },
  postComment: jest.fn(),
  downVoteRecipe: jest.fn(),
  addOrRemoveFavourite: jest.fn(),
  fetchRecipeComment: jest.fn(),
  upVoteRecipe: jest.fn(),
  fetchRecipe: jest.fn()
};

const state = {
  isLoading: false,
  comment: '',
  errors: {}
};

describe('<SingleRecipe />', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<SingleRecipe {...props} {...state} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<SingleRecipe />', () => {
  it('should call componentWillReceiveProps', () => {
    const wrapper = shallow(<SingleRecipe {...props} />);
    const componentWillReceivePropsSpy = jest
      .spyOn(wrapper.instance(), 'componentWillReceiveProps');
    wrapper.update();
    wrapper.setProps({
      ...props,
      recipeReducer: {
        isFetched: true,
        recipes: mockData.recipe,
      },
      commentReducer: {
        reviews: mockData.reviews
      },
    });
    expect(componentWillReceivePropsSpy).toHaveBeenCalled();
  });
});

describe('<SingleRecipe />', () => {
  it('should call handleUpvote function', () => {
    const wrapper = shallow(<SingleRecipe {...props} {...state} />);
    const handleUpvoteSpy = jest.spyOn(wrapper.instance(), 'handleUpvote');
    wrapper.instance().handleUpvote();
    expect(handleUpvoteSpy).toHaveBeenCalled();
    expect(wrapper.instance().props.upVoteRecipe).toHaveBeenCalled();
  });
});

describe('<SingleRecipe />', () => {
  it('should call handleDownvote function', () => {
    const wrapper = shallow(<SingleRecipe {...props} {...state} />);
    const handleDownvoteSpy = jest.spyOn(wrapper.instance(), 'handleDownvote');
    wrapper.instance().handleDownvote();
    expect(handleDownvoteSpy).toHaveBeenCalled();
    expect(wrapper.instance().props.downVoteRecipe).toHaveBeenCalled();
  });
});

describe('<SingleRecipe />', () => {
  it('should call handleAddToFavourite function', () => {
    const wrapper = shallow(<SingleRecipe {...props} {...state} />);
    const handleAddToFavouriteSpy = jest
      .spyOn(wrapper.instance(), 'handleAddToFavourite');
    wrapper.instance().handleAddToFavourite();
    expect(handleAddToFavouriteSpy).toHaveBeenCalled();
    expect(wrapper.instance().props.addOrRemoveFavourite).toHaveBeenCalled();
  });
});
