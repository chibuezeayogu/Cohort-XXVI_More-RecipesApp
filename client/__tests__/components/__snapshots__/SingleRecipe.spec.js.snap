// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleRecipe /> should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body grey lighten-5"
  >
    <withRouter(Connect(UserMenu))
      addOrRemoveFavourite={[MockFunction]}
      comment=""
      commentReducer={
        Object {
          "reviews": Array [
            Object {
              "User": Object {
                "firstName": "Chibueze",
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                "lastName": "Nelson Ayogu",
              },
              "comment": "dfasdfaasdfadsf",
              "createdAt": "2018-02-02T18:59:23.176Z",
              "id": 22,
              "recipeId": 1,
              "updatedAt": "2018-02-02T18:59:23.176Z",
              "userId": 1,
              "view": false,
            },
            Object {
              "User": Object {
                "firstName": "Chibueze",
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                "lastName": "Nelson Ayogu",
              },
              "comment": "dfasdfa",
              "createdAt": "2018-02-02T18:59:18.411Z",
              "id": 21,
              "recipeId": 1,
              "updatedAt": "2018-02-02T18:59:18.411Z",
              "userId": 1,
              "view": false,
            },
          ],
        }
      }
      downVoteRecipe={[MockFunction]}
      errors={Object {}}
      fetchRecipe={
        [MockFunction] {
          "calls": Array [
            Array [
              1,
            ],
          ],
        }
      }
      fetchRecipeComment={
        [MockFunction] {
          "calls": Array [
            Array [
              1,
            ],
          ],
        }
      }
      isLoading={false}
      match={
        Object {
          "params": Object {
            "id": 1,
          },
        }
      }
      postComment={[MockFunction]}
      recipeReducer={
        Object {
          "isFetched": true,
          "recipes": Array [
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-28T22:46:03.051Z",
              "description": "asdfafd",
              "downvotes": 0,
              "id": 1,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "adfadfadf; adfadfadf",
              "procedures": "adfafadf; adfadfadf",
              "title": "CHANGED",
              "updatedAt": "2018-02-06T19:28:36.149Z",
              "upvotes": 1,
              "views": 0,
            },
          ],
        }
      }
      upVoteRecipe={[MockFunction]}
      userData={
        Object {
          "currentUser": Object {
            "email": "chibuezeayogu@hotmail.com",
            "id": 1,
          },
        }
      }
    />
    <div
      className="main"
    >
      <div
        className="container"
      >
        <Preloader />
      </div>
    </div>
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body grey lighten-5"
    >
      <withRouter(Connect(UserMenu))
        addOrRemoveFavourite={[MockFunction]}
        comment=""
        commentReducer={
          Object {
            "reviews": Array [
              Object {
                "User": Object {
                  "firstName": "Chibueze",
                  "id": 1,
                  "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                  "lastName": "Nelson Ayogu",
                },
                "comment": "dfasdfaasdfadsf",
                "createdAt": "2018-02-02T18:59:23.176Z",
                "id": 22,
                "recipeId": 1,
                "updatedAt": "2018-02-02T18:59:23.176Z",
                "userId": 1,
                "view": false,
              },
              Object {
                "User": Object {
                  "firstName": "Chibueze",
                  "id": 1,
                  "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                  "lastName": "Nelson Ayogu",
                },
                "comment": "dfasdfa",
                "createdAt": "2018-02-02T18:59:18.411Z",
                "id": 21,
                "recipeId": 1,
                "updatedAt": "2018-02-02T18:59:18.411Z",
                "userId": 1,
                "view": false,
              },
            ],
          }
        }
        downVoteRecipe={[MockFunction]}
        errors={Object {}}
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        fetchRecipeComment={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        isLoading={false}
        match={
          Object {
            "params": Object {
              "id": 1,
            },
          }
        }
        postComment={[MockFunction]}
        recipeReducer={
          Object {
            "isFetched": true,
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf; adfadfadf",
                "procedures": "adfafadf; adfadfadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          }
        }
        upVoteRecipe={[MockFunction]}
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
          }
        }
      />
      <div
        className="main"
      >
        <div
          className="container"
        >
          <Preloader />
        </div>
      </div>
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SingleRecipe
        addOrRemoveFavourite={[MockFunction]}
        comment=""
        commentReducer={
          Object {
            "reviews": Array [
              Object {
                "User": Object {
                  "firstName": "Chibueze",
                  "id": 1,
                  "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                  "lastName": "Nelson Ayogu",
                },
                "comment": "dfasdfaasdfadsf",
                "createdAt": "2018-02-02T18:59:23.176Z",
                "id": 22,
                "recipeId": 1,
                "updatedAt": "2018-02-02T18:59:23.176Z",
                "userId": 1,
                "view": false,
              },
              Object {
                "User": Object {
                  "firstName": "Chibueze",
                  "id": 1,
                  "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                  "lastName": "Nelson Ayogu",
                },
                "comment": "dfasdfa",
                "createdAt": "2018-02-02T18:59:18.411Z",
                "id": 21,
                "recipeId": 1,
                "updatedAt": "2018-02-02T18:59:18.411Z",
                "userId": 1,
                "view": false,
              },
            ],
          }
        }
        downVoteRecipe={[MockFunction]}
        errors={Object {}}
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        fetchRecipeComment={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        isLoading={false}
        match={
          Object {
            "params": Object {
              "id": 1,
            },
          }
        }
        postComment={[MockFunction]}
        recipeReducer={
          Object {
            "isFetched": true,
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf; adfadfadf",
                "procedures": "adfafadf; adfadfadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          }
        }
        upVoteRecipe={[MockFunction]}
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SingleRecipe {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "handleDownvote": [Function],
        "handleOnsubmit": [Function],
        "props": Object {
          "addOrRemoveFavourite": [MockFunction],
          "comment": "",
          "commentReducer": Object {
            "reviews": Array [
              Object {
                "User": Object {
                  "firstName": "Chibueze",
                  "id": 1,
                  "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                  "lastName": "Nelson Ayogu",
                },
                "comment": "dfasdfaasdfadsf",
                "createdAt": "2018-02-02T18:59:23.176Z",
                "id": 22,
                "recipeId": 1,
                "updatedAt": "2018-02-02T18:59:23.176Z",
                "userId": 1,
                "view": false,
              },
              Object {
                "User": Object {
                  "firstName": "Chibueze",
                  "id": 1,
                  "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                  "lastName": "Nelson Ayogu",
                },
                "comment": "dfasdfa",
                "createdAt": "2018-02-02T18:59:18.411Z",
                "id": 21,
                "recipeId": 1,
                "updatedAt": "2018-02-02T18:59:18.411Z",
                "userId": 1,
                "view": false,
              },
            ],
          },
          "downVoteRecipe": [MockFunction],
          "errors": Object {},
          "fetchRecipe": [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          },
          "fetchRecipeComment": [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          },
          "isLoading": false,
          "match": Object {
            "params": Object {
              "id": 1,
            },
          },
          "postComment": [MockFunction],
          "recipeReducer": Object {
            "isFetched": true,
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf; adfadfadf",
                "procedures": "adfafadf; adfadfadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          },
          "upVoteRecipe": [MockFunction],
          "userData": Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
          },
        },
        "refs": Object {},
        "renderComment": [Function],
        "state": Object {
          "comment": "",
          "errors": Object {},
          "isLoading": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body grey lighten-5"
        >
          <withRouter(Connect(UserMenu))
            addOrRemoveFavourite={[MockFunction]}
            comment=""
            commentReducer={
              Object {
                "reviews": Array [
                  Object {
                    "User": Object {
                      "firstName": "Chibueze",
                      "id": 1,
                      "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                      "lastName": "Nelson Ayogu",
                    },
                    "comment": "dfasdfaasdfadsf",
                    "createdAt": "2018-02-02T18:59:23.176Z",
                    "id": 22,
                    "recipeId": 1,
                    "updatedAt": "2018-02-02T18:59:23.176Z",
                    "userId": 1,
                    "view": false,
                  },
                  Object {
                    "User": Object {
                      "firstName": "Chibueze",
                      "id": 1,
                      "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                      "lastName": "Nelson Ayogu",
                    },
                    "comment": "dfasdfa",
                    "createdAt": "2018-02-02T18:59:18.411Z",
                    "id": 21,
                    "recipeId": 1,
                    "updatedAt": "2018-02-02T18:59:18.411Z",
                    "userId": 1,
                    "view": false,
                  },
                ],
              }
            }
            downVoteRecipe={[MockFunction]}
            errors={Object {}}
            fetchRecipe={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            fetchRecipeComment={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            isLoading={false}
            match={
              Object {
                "params": Object {
                  "id": 1,
                },
              }
            }
            postComment={[MockFunction]}
            recipeReducer={
              Object {
                "isFetched": true,
                "recipes": Array [
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-28T22:46:03.051Z",
                    "description": "asdfafd",
                    "downvotes": 0,
                    "id": 1,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "adfadfadf; adfadfadf",
                    "procedures": "adfafadf; adfadfadf",
                    "title": "CHANGED",
                    "updatedAt": "2018-02-06T19:28:36.149Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                ],
              }
            }
            upVoteRecipe={[MockFunction]}
            userData={
              Object {
                "currentUser": Object {
                  "email": "chibuezeayogu@hotmail.com",
                  "id": 1,
                },
              }
            }
          />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <Preloader />
            </div>
          </div>
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body grey lighten-5"
        >
          <withRouter(Connect(UserMenu))
            addOrRemoveFavourite={[MockFunction]}
            comment=""
            commentReducer={
              Object {
                "reviews": Array [
                  Object {
                    "User": Object {
                      "firstName": "Chibueze",
                      "id": 1,
                      "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                      "lastName": "Nelson Ayogu",
                    },
                    "comment": "dfasdfaasdfadsf",
                    "createdAt": "2018-02-02T18:59:23.176Z",
                    "id": 22,
                    "recipeId": 1,
                    "updatedAt": "2018-02-02T18:59:23.176Z",
                    "userId": 1,
                    "view": false,
                  },
                  Object {
                    "User": Object {
                      "firstName": "Chibueze",
                      "id": 1,
                      "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
                      "lastName": "Nelson Ayogu",
                    },
                    "comment": "dfasdfa",
                    "createdAt": "2018-02-02T18:59:18.411Z",
                    "id": 21,
                    "recipeId": 1,
                    "updatedAt": "2018-02-02T18:59:18.411Z",
                    "userId": 1,
                    "view": false,
                  },
                ],
              }
            }
            downVoteRecipe={[MockFunction]}
            errors={Object {}}
            fetchRecipe={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            fetchRecipeComment={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            isLoading={false}
            match={
              Object {
                "params": Object {
                  "id": 1,
                },
              }
            }
            postComment={[MockFunction]}
            recipeReducer={
              Object {
                "isFetched": true,
                "recipes": Array [
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-28T22:46:03.051Z",
                    "description": "asdfafd",
                    "downvotes": 0,
                    "id": 1,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "adfadfadf; adfadfadf",
                    "procedures": "adfafadf; adfadfadf",
                    "title": "CHANGED",
                    "updatedAt": "2018-02-06T19:28:36.149Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                ],
              }
            }
            upVoteRecipe={[MockFunction]}
            userData={
              Object {
                "currentUser": Object {
                  "email": "chibuezeayogu@hotmail.com",
                  "id": 1,
                },
              }
            }
          />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <Preloader />
            </div>
          </div>
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SingleRecipe
    addOrRemoveFavourite={[MockFunction]}
    comment=""
    commentReducer={
      Object {
        "reviews": Array [
          Object {
            "User": Object {
              "firstName": "Chibueze",
              "id": 1,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
              "lastName": "Nelson Ayogu",
            },
            "comment": "dfasdfaasdfadsf",
            "createdAt": "2018-02-02T18:59:23.176Z",
            "id": 22,
            "recipeId": 1,
            "updatedAt": "2018-02-02T18:59:23.176Z",
            "userId": 1,
            "view": false,
          },
          Object {
            "User": Object {
              "firstName": "Chibueze",
              "id": 1,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517772638/oqbysjf7xgfv2dko409s.jpg",
              "lastName": "Nelson Ayogu",
            },
            "comment": "dfasdfa",
            "createdAt": "2018-02-02T18:59:18.411Z",
            "id": 21,
            "recipeId": 1,
            "updatedAt": "2018-02-02T18:59:18.411Z",
            "userId": 1,
            "view": false,
          },
        ],
      }
    }
    downVoteRecipe={[MockFunction]}
    errors={Object {}}
    fetchRecipe={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
      }
    }
    fetchRecipeComment={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
      }
    }
    isLoading={false}
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    postComment={[MockFunction]}
    recipeReducer={
      Object {
        "isFetched": true,
        "recipes": Array [
          Object {
            "addedBy": 1,
            "createdAt": "2018-01-28T22:46:03.051Z",
            "description": "asdfafd",
            "downvotes": 0,
            "id": 1,
            "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
            "ingredients": "adfadfadf; adfadfadf",
            "procedures": "adfafadf; adfadfadf",
            "title": "CHANGED",
            "updatedAt": "2018-02-06T19:28:36.149Z",
            "upvotes": 1,
            "views": 0,
          },
        ],
      }
    }
    upVoteRecipe={[MockFunction]}
    userData={
      Object {
        "currentUser": Object {
          "email": "chibuezeayogu@hotmail.com",
          "id": 1,
        },
      }
    }
  />,
}
`;
